### PR TITLE
Remove macOS sigtramp hack

### PR DESCRIPTION
**Pull request by benesch**
Since Go 1.11, Go no longer abuses the internal sigtramp field on macOS
when it installs its signal handlers [0]. This means that we no longer
need several layers of deviousness to restore Go's signal handlers on
Darwin.

Once this commit lands, we should add a note to the README indicating
that users who want support for Go versions early than 1.11 will need to
use the commit prior to this one.

[0]: https://github.com/golang/go/commit/ec989337c5d3203d52de1a2314813996c711fce6

### CODE REVIEW

**Review by knz — COMMENTED** [review]
I'm ok for dropping the old go compatibility, however please add some `.go` file with `// +build !go1.12` and an `init()` function that panics, to make it clear to the user what they need to do.

**Comment by benesch** [comment]
Sure, done! PTAL.

File: version_check.go [hunk 1]
```diff
@@ -12,12 +12,10 @@
...
 
-package sigtramp
+package libedit
 
-import "unsafe"
-
-func Get() unsafe.Pointer {
-	return unsafe.Pointer(uintptr(0))
+func init() {
+  panic("go-libedit requires go1.11+")
```
Comment (knz) [inline version_check.go:20]: you mean go1.12 in the error message

**Review by knz — APPROVED** [review]
LGTM with nit

File: version_check.go (hunk 1 above)
Comment (benesch) [reply version_check.go:20]: I don't think so? The signal handling changes landed in go1.11, unless I'm mistaken: https://github.com/golang/go/commit/ec989337c5d3203d52de1a2314813996c711fce6

Comment (knz) [reply version_check.go:20]: oops, yes, you're right. merge away :)

**Pull request merged**